cpp.md

```diff
@@ -379,43 +434,203 @@ int main() {
+		cout << ' ' << pq.top();
+		pq.pop();
+	}
+}
 
-int arr[] = {3,2,1};
-int n = 3 ; // size of the array
+```
 
-sort(arr, arr+n); // sorting the array
+### Set
 
-vector<int> v;
+Sets are a type of container in which each element has to be unique. The values are stored in a specific sorted order.
 
-v = {3,2,1};
+insert(g) - Adds a new element ‘g’ to the set.
+erase(g) - Removes the value ‘g’ from the set.
+size() – Returns the number of elements in the set.
+empty() – Returns whether the set is empty.
+begin() – Returns an iterator to the first element in the set.
+end() – Returns an iterator to the element that follows the last element in the set.
 
-sort(v.begin(), v.end()); // sorting in the vector
+```c
+#include <iostream>
+#include <set>
+using namespace std;
+int main() {
+	set<int> st;
+	st.insert(40);
+    st.insert(30);
+    st.insert(60);
+    st.insert(20);
+    st.insert(50);
+
+    // printing set st
+	for (i = st.begin(); i != st.end(); i++) {
+        cout << *i << " ";
+    }
 
+    //another way to print
+    for (auto i: st) {
+        cout << i << " ";
+    }
 }
 
 ```
-## Header Files
 
-|Name| Use|
-|----|----|
-|#include<stdio.h>   | It is used to perform input and output operations|
-|#include<string.h>  | It is used to perform various string operations|
-|#include<math.h>    | It is used to perform mathematical operations|
-|#include<iomanip.h> | It is used to access set() and setprecision()|
-|#include<signal.h>  | It is used to perform signal handling functions like sign|
-|#include<stdarg.h>  | It is used to perform standard argument functions|
-|#include<errno.h>   | It is used to perform error handling operations like errno|
-|#include<fstream.h> | It is used to control the data to read from a file|
-|#include<time.h>    | It is used to perform functions related to date() and time|
+### Map
+
+Maps are type of containers that store elements in a mapped fashion. Each element has a key value and a mapped value. No two mapped values can have the same key values.
+
+insert(keyvalue, mapvalue) – Adds a new element to the map.
+erase(g)– Removes the key-value ‘g’ from the map.
+begin() – Returns an iterator to the first element in the map.
+end() – Returns an iterator to the element that follows the last element in the map.
+size() – Returns the number of elements in the map.
+empty() – Returns whether the map is empty.
+clear() – Removes all the elements from the map.
+
+```c
+#include <iostream>
+#include <map>
+using namespace std;
+int main() {
+	map<int, int> mp;
+	mp.insert({1, 40});
+    mp.insert({2, 30});
+    mp.insert({3, 60});
+    mp.insert({4, 20});
+    mp.insert({5, 50});
+
+    mp[6] = 10;   // another way of inserting a value in a map
+
+    // printing map mp
+	for (i = mp.begin(); i != mp.end(); i++) {
+        cout << " " << i->first << " " << i->second << endl;
+    }
+
+    for (auto i: mp) {
+        cout << " " << i.first << " " << i.second << endl;
+    }
+}
+
+```
+
+## Algorithms
```

Comment:
Can you make this as a Markdown table to represent in a concise way?

Reply:
@onecompiler-ops sure why not! I guess I had made it lengthy !